framework/src/Volo.Abp.DistributedLocking.Dapr/Volo/Abp/DistributedLocking/Dapr/DaprAbpDistributedLock.cs

```diff
@@ -0,0 +1,50 @@
+﻿using Microsoft.Extensions.Options;
+using Volo.Abp.Dapr;
+using Volo.Abp.DependencyInjection;
+
+namespace Volo.Abp.DistributedLocking.Dapr;
+
+[Dependency(ReplaceServices = true)]
+public class DaprAbpDistributedLock : IAbpDistributedLock, ITransientDependency
+{
+    protected AbpDaprClientFactory DaprClientFactory { get; }
+    protected AbpDistributedLockDaprOptions DistributedLockDaprOptions { get; }
+    protected AbpDaprOptions DaprOptions { get; }
+    
+    public DaprAbpDistributedLock(
+        AbpDaprClientFactory daprClientFactory,
+        IOptions<AbpDistributedLockDaprOptions> distributedLockDaprOptions,
+        IOptions<AbpDaprOptions> daprOptions)
+    {
+        DaprClientFactory = daprClientFactory;
+        DaprOptions = daprOptions.Value;
+        DistributedLockDaprOptions = distributedLockDaprOptions.Value;
+    }
+    
+    public async Task<IAbpDistributedLockHandle> TryAcquireAsync(
+        string name,
+        TimeSpan timeout = default,
+        CancellationToken cancellationToken = default)
+    {
+        if (timeout == default)
+        {
+            timeout = DistributedLockDaprOptions.DefaultTimeout;
+        }
+        
+        var daprClient = await DaprClientFactory.CreateAsync();
```

Comment:
Is DaprClient disposable? Shouldn't we dispose the client after lock is released (or if can't be obtained)? Also check other places we used the DaprClient. Also, is DaprClient thread-safe and reusable (I don't know now)?

Reply:
https://github.com/dapr/dotnet-sdk/blob/016545ac0d5c114b72795e187069d21f0a333048/src/Dapr.Client/TryLockResponse.cs

DaprClient is thread-safe and reusable, This is also used by the Dapr sdk